src/transport_ofi.c

```diff
@@ -1211,24 +1283,104 @@ int shmem_transport_startup(void)
     return 0;
 }
 
-void shmem_transport_ctx_create(shmem_transport_ctx_t **ctx)
+int shmem_transport_ctx_create(shmem_transport_ctx_t **ctx)
+{
+    SHMEM_MUTEX_LOCK(shmem_transport_ofi_lock);
+
+    int ret;
+    int id = shmem_transport_ofi_num_ctx;
+
+    /* FIXME: This does not do resource cleanup 
+     * on error, it just returns. This is consistent with how
+     * initialization worked before, but is worse since context creation
+     * is not necessarily do-or-die.
+     */
+    /* JD: This needs to have a mutex around it to synchronize a possible
+     * update to the contexts array.
+     *
+     * This also does not reuse entries from contexts that were freed.
+     */
```

Comment:
The part about the mutex can be deleted now that this is guarded by ```shmem_transport_ofi_lock```, but the part about leaking entries is still a valid FIXME.